### PR TITLE
Release 1.5.0

### PR DESCRIPTION
**Pull request by markmandel**
Tweaked the ordering in the release checklist closer to what I usually do as well.

### CODE REVIEW

**Comment by aLekSer** [comment]
Just for curiosity's sake: In the release process, where goes the step `make build-images && make push`?

File: site/content/en/docs/Reference/gameserver.md [hunk 1]
```diff
@@ -183,27 +108,6 @@ The `spec` field is the actual GameServer specification and it is composed as fo
   - `grpcPort` the port that the SDK Server binds to for gRPC connections
   - `httpPort` the port that the SDK Server binds to for HTTP gRPC gateway connections
 - `template` the [pod spec template](https://v1-13.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#podtemplatespec-v1-core) to run your GameServer containers, [see](https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/#pod-templates) for more information.
```
Comment (aLekSer) [inline site/content/en/docs/Reference/gameserver.md:110]: Should be more recent version:
https://v1-14.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/

**Comment by aLekSer** [comment]
@markmandel  please look through the `md` files if they still contains old version of Lubernetes docs. I got such an error in chrome and `Not found` page if proceed with wrong certificate:
```
v1-13.docs.kubernetes.io 
NET::ERR_CERT_COMMON_NAME_INVALID
```

**Comment by markmandel** [comment]
Nice catch! Fixed.

**Comment by markmandel** [comment]
> Just for curiosity's sake: In the release process, where goes the step make build-images && make push?

All part of `make do-release`

**Comment by aLekSer** [comment]
I understood why `make hugo-test` did not catch `http://gcr.io/agones-images/agones-controller:1.5.0` - the image is absent, but the link is working.

**Pull request merged**